reverse_proxy/default.conf

```diff
@@ -27,6 +27,44 @@ server {
     gzip_comp_level 9;
 
 
+    location /frontend/ {
+        add_header 'Access-Control-Allow-Origin' '*';
+        add_header 'Access-Control-Expose-Headers' '*';
+        add_header 'Access-Control-Allow-Credentials' 'true';
+        add_header 'Access-Control-Allow-Methods' 'GET, POST, OPTIONS';
+        add_header 'Access-Control-Allow-Headers' 'DNT,X-CustomHeader,Keep-Alive,User-Agent,X-Requested-With,If-Modified-Since,Cache-Control,Content-Type';
+        proxy_pass http://frontend/;
+        proxy_set_header X-Forwarded-Host $host;
+        proxy_set_header X-Forwarded-Proto $scheme;
+        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
+    }
+
+    
+    location /light/ {
+        add_header 'Access-Control-Allow-Origin' '*';
+        add_header 'Access-Control-Expose-Headers' '*';
+        add_header 'Access-Control-Allow-Credentials' 'true';
+        add_header 'Access-Control-Allow-Methods' 'GET, POST, OPTIONS';
+        add_header 'Access-Control-Allow-Headers' 'DNT,X-CustomHeader,Keep-Alive,User-Agent,X-Requested-With,If-Modified-Since,Cache-Control,Content-Type';
+        proxy_pass http://compare-frontend/;
+        proxy_set_header X-Forwarded-Host $host;
+        proxy_set_header X-Forwarded-Proto $scheme;
+        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
+    }
+
+    location /backend/ {
+        add_header 'Access-Control-Allow-Origin' '*';
+        add_header 'Access-Control-Expose-Headers' '*';
+        add_header 'Access-Control-Allow-Credentials' 'true';
+        add_header 'Access-Control-Allow-Methods' 'GET, POST, OPTIONS';
```

Comment:
And here the question if there is the risk that patch or delete requests needs to be supported from a different host. Currently we only allow GET, POST & OPTIONS requests. For the wps & the geoserver those are ok, as we don't expect external clients to patch or delete (we oursleves do run those requests, but then directly in the docker network, so that cors is not an issue).

However, for your backend the situation may look different...

Reply:
Quite right!
The `backend` container only expects POST and GET requests, and OPTIONs when the browser needs to run those. No PUTs, PATCHes or DELETEs there; fortunately. In fact, the backend-api is fairly simple, using only 4 GET endpoints and 1 POST endpoint : https://github.com/riesgos/dlr-riesgos-frontend/blob/main/backend/src/scenarios/scenario.interface.ts